geometrycollection.go

```diff
@@ -7,6 +7,9 @@ type GeometryCollection struct {
 	srid  int
 }
 
+// GeometryCollection implements interface T.
+var _ T = (*GeometryCollection)(nil)
```

Comment:
Please move the interface satisfaction tests to `geom_test.go` and use the style:

```go
var _ T = &GeometryCollection{}
```